schema/tag.js

```diff
@@ -34,7 +41,20 @@ const Tag = {
       type: new GraphQLNonNull(GraphQLString),
     },
   },
-  resolve: (root, { id }) => gravity(`tag/${id}`),
+  resolve: (root, { id }, request, { fieldNodes }) => {
+    // If you are just making an artworks call ( e.g. if paginating )
+    // do not make a Gravity call for the gene data.
+    const blacklistedFields = ["filtered_artworks", "id", "__id"]
+    if (queriedForFieldsOtherThanBlacklisted(fieldNodes, blacklistedFields)) {
+      return gravity(`tag/${id}`).then(tag => {
+        return Object.assign(tag, { _type: "Tag" }, {})
```

Comment:
cc @joeyAghion at the time of fetching the tag, we _know_ it's a tag, so include that here.

Reply:
Ahh, but see, the 👹 is in the details. The requirements of `isTypeOf` is such that the type can be identified for a payload that comes from outside a closely bound field like this, which would mostly come into play when we’d have feeds that can contain mixed types as long as they implement a common interface. Seeing as we don’t actually have any of those atm, the conclusion was that we should probably just move to using `resolveType` on the interface, which is closely bound to the data fetching for that interface and where we can guarantee that the type will be injected into the payload.